src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -269,6 +269,13 @@ namespace AppInstaller::CLI::Workflow
                 (isUpdate ? MSStoreUpdate : MSStoreInstall);
             break;
         case InstallerTypeEnum::Portable:
+            if (isUpdate && installer.UpdateBehavior == UpdateBehaviorEnum::UninstallPrevious)
+            {
+                context <<
+                    GetUninstallInfo <<
+                    ExecuteUninstaller;

```

Comment:
I think the upgrade flow for portable apps might need to be a little bit more complex. Based on running an uninstall on upgrade, with the current logic, you're removing the executable, the install directory, and the symlink. If an upgrade is being performed, wouldn't it make more sense to leave the install directory and symlink, replacing just the binary itself?

Reply:
One of the challenges is that the binary might still be running.

Reply:
> One of the challenges is that the binary might still be running.

Wouldn't that be a challenge either way? The executable wouldn't be removed if it were still running regardless of if it is just the executable being removed, or the entire structure of executable, directory, and symlink. I think failing to delete the executable while it is still running makes sense, and is what I would expect. 

To clarify, I'm not saying that the binary should be overwritten without an attempted deletion first. I'm just noting that in an upgrade scenario, it doesn't seem logical to me to remove the folder and the symlink just to create them again when the app is installed

Reply:
After processing the change a little more, it might already be doing `upgrade` as "install over top of".  I think that the current pattern isn't quite forward looking for portable archives, and I'm not sure that upgrade will be as clean as install got to be for being able to remove at any point in the process.

Reply:
The uninstall flow is only invoked if the upgrade behavior is set to `uninstallPrevious`. If that is not present, then the upgrade will call the regular install flow. I have changed the install flow so that it replaces the files that needs to be updated and writes that change to registry. If the upgrade fails at any point, the existing version is preserved.